File: DanceSequence/Models/Views/MoveDetails.cs
using System.Collections.Generic;

namespace DanceSequence.Models.Views
{
    public class MoveDetails
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public List<MoveView> PreMoves { get; set; }
        public List<MoveView> ProMoves { get; set; }
        public List<AlternationView> Alternations { get; set; }
        public List<TagView> Tags { get; set; }
    }
}

File: DanceSequence/ClientApp/src/app/model/moveDetails.ts
import { Move } from '../model/move';
import { Tag } from '../model/tag';
import { Alternation } from '../model/alternation';

export class MoveDetails {
    public id: number = 0;
    public name: string;
    public description: string;
    public preMoves: Move[];
    public proMoves: Move[];
    public alternations: Alternation[];
    public tags: Tag[];
}